### PR TITLE
Add surrogate-key caching for all location-based queries

### PR DESCRIPTION
**Pull request by opyh**
This improves caching for tile- and radius-based queries.

Documentation: https://beta.observablehq.com/@opyh/speed-up-radius-based-geo-queries-over-cdn

### CODE REVIEW

**Comment by mutaphysis** [comment]
Very cool!

File: both/api/shared/tile-indexing/tileSurrogateKeysForFeature.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (
+    !feature ||
+    !feature.geometry ||
+    feature.geometry.type !== 'Point' ||
+    !feature.geometry.coordinates
+  ) {
+    return [];
+  }
+  return Array.from({ length: 22 }).map((_, z) => {
```
[the remark this answers is not in the document]
Comment (opyh) [reply both/api/shared/tile-indexing/tileSurrogateKeysForFeature.ts]: We support level 21 in our API and Mapbox too -> changed to 21

File: both/api/shared/tile-indexing/getTilesEnclosingCircle.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { latlon2tile } from './latlon2tile';
+
+/**
+ * @returns A new `LatLon` that is offset by the given coordinates in meters (with an error margin
+ *   of a few meters that depends on the latitude)
+ */
+function offsetLatLon({ lat, lon }: LatLon, offsetInMeters: Point2D): LatLon {
+  const earthRadiusInMeters = 6378137;
+
+  const δLat = (offsetInMeters.y / earthRadiusInMeters)
```
[the remark this answers is not in the document]
Comment (opyh) [reply both/api/shared/tile-indexing/getTilesEnclosingCircle.ts]: Changed `δ` -> `offset`

[the remark this answers is not in the document]
Comment (opyh) [reply both/api/shared/tile-indexing/getTilesEnclosingCircle.ts]: (Will just use emojis next time) ;)

**Pull request merged**